app/helpers/Intersect.js

```diff
@@ -571,3 +572,41 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+async function getPageData(authority, page) {
+  // this function should always return an object.
+  // if there's no data, return an empty object.
+  const cacheKey = authority.internal_id + '|page:' + page.id;
```

Comment:
Should the key include some timestamp to ensure we have a recent version?

Reply:
I'm not sure if I understood your comment.

If we append current timestamp to the key, it'll cause a cache miss every single time.

If we want to check on recent version, are you suggesting we store the saved timestamp in the payload and we check for expiration?

Reply:
Yes, I was thinking more of invalidation logic from the producer of the pages, a version number in the payload would make that simpler but I am not aware of how much control we have on what comes back. So maybe we can fabricate a timestamp with a `reasonable` window for invalidating our cache, I don't know how long that would be given the nature of the content in these pages.

Reply:
Yeah, I think cache invalidation has to come from the server, as any invalidation logic we put in would be quite arbitrary. We had a discussion with the backend team, and they're planning on adding a hash for invalidation. When that happens, we can add the hash to the cache key which will be better than any timestamp right now. So we can go with what I have currently and add the hash in whenever it's available, how does that sound?